scripts/process-data/healthkitv2electrocardiogram.R

```diff
@@ -0,0 +1,216 @@
+      dplyr::ungroup() %>% 
+      tidyr::pivot_longer(cols = c(numrecords), 
+                          names_to = "stat", 
+                          values_to = "value") %>% 
+      dplyr::mutate(concept = paste0("mhp:summary:weekly:", stat, ":", concept)) %>% 
+      dplyr::select(participantidentifier, startdate, enddate, concept, value) %>% 
+      dplyr::distinct()
+  }
+  
+  result <- 
+    dplyr::bind_rows(summarize_alltime(df), 
+                     summarize_weekly(df)) %>% 
+    dplyr::distinct()
+  
+  return(result)
+}
+
+library(dplyr)
+
+dataset <- "healthkitv2electrocardiogram"
+
+cat(glue::glue("Transforming data for {dataset}"),"\n")
+
+# Get variables for this dataset
+vars <- 
+  selected_vars %>% 
+  filter(grepl(paste0(dataset, "$"), Export, ignore.case = TRUE, perl = TRUE)) %>% 
+  pull(Variable)
+
+# Exclude participants who have i2b2 summaries from fitbit data
+participants_to_exclude <- 
+  read.csv(file.path(outputConceptsDir, "fitbit_participants.csv")) %>% 
+  pull(participantidentifier)
+
+# Load the desired subset of this dataset in memory and do some feature engineering for derived variables
+df <- 
+  arrow::open_dataset(file.path(downloadLocation, glue::glue("dataset_{dataset}"))) %>% 
+  select(all_of(c(vars))) %>% 
+  dplyr::filter(!(ParticipantIdentifier %in% participants_to_exclude)) %>% 
+  filter(Classification %in% c("SinusRhythm", "AtrialFibrillation")) %>% 
+  collect() %>% 
+  rename(EndDate = Date)
+
+colnames(df) <- tolower(colnames(df))
+
+# Create lists for ID variables and i2b2 concept variables
+excluded_concepts <- c("participantidentifier", "startdate", "enddate", "date")
+
+approved_concepts_summarized <- 
+  setdiff(
+    tolower(selected_vars$Variable[selected_vars$Export==dataset]),
+    excluded_concepts
+  )
+
+# Pivot data frame from long to wide
+df_melted_filtered <- 
+  df %>% 
+  mutate("SinusRhythm" = if_else(classification == "SinusRhythm", 1, NA),
+         "AtrialFibrillation" = if_else(classification == "AtrialFibrillation", 1, NA)) %>% 
+  select(-c(classification)) %>% 
+  recoverSummarizeR::melt_df(excluded_concepts = excluded_concepts) %>% 
+  select(if("participantidentifier" %in% colnames(.)) "participantidentifier",
+         dplyr::matches("(?<!_)date(?!_)", perl = T),
+         if("concept" %in% colnames(.)) "concept",
+         if("value" %in% colnames(.)) "value") %>% 
+  tidyr::drop_na("value") %>% 
+  mutate(value = as.numeric(value))
+cat("recoverSummarizeR::melt_df() completed.\n")
+
+# Generate i2b2 summaries
+df_summarized <- 
+  df_melted_filtered %>% 
+  select(all_of(c("participantidentifier", "startdate", "enddate", "concept", "value"))) %>% 
+  ecg_stat_summarize() %>% 
+  mutate(value = as.numeric(value)) %>% 
+  distinct()
+cat("recoverSummarizeR::stat_summarize() completed.\n")
+
+# Add i2b2 columns from concept map (ontology file) and clean the output
+output_concepts <- 
+  process_df(df_summarized, 
```

Comment:
Does this need recoverSummarizeR::  in front?

Reply:
Fixed in commit [5a92640](https://github.com/Sage-Bionetworks/recover-pipeline-i2b2/pull/21/commits/5a92640b240006fdd1671dc765faef729c459221)